src/content/en/fundamentals/primers/modules.md

```diff
@@ -0,0 +1,578 @@
+other browsers. The ability to make this distinction is amazing, if only for performance! Think
+about it: only modern browsers support modules. If a browser understands your module code, it also
+supports [features that were around before modules](https://codepen.io/samthor/pen/MmvdOM), such as
+arrow functions or `async`-`await`. You don’t have to transpile those features in your module
+bundle anymore! You can [serve smaller and largely untranspiled module-based payloads to modern
+browsers](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/). Only
+legacy browsers get the `nomodule` payload.
+
+### Browser-specific differences between modules and classic scripts
+
+As you now know, modules are different from classic scripts. On top of the platform-agnostic
+differences we’ve outlined above, there are some differences that are specific to browsers.
+
+For example, modules are evaluated only once, while classic scripts are evaluated however many
+times you add them to the DOM.
+
+```js
+<script src="classic.js"></script>
+<script src="classic.js"></script>
+<!-- classic.js executes multiple times. -->
+
+<script type="module" src="module.mjs"></script>
+<script type="module" src="module.mjs"></script>
+<script type="module">import './module.mjs';</script>
+<!-- module.mjs executes only once. -->
+```
+
+Also, module scripts and their dependencies are fetched with CORS. This means that any
+cross-origin module scripts must be served with the proper headers, such as
+`Access-Control-Allow-Origin: *`. This is not true for classic scripts.
+
+Another difference relates to the `async` attribute, which causes the script to download without
+blocking the HTML parser (like `defer`) except it also executes the script as soon as possible,
+with no guaranteed order, and without waiting for HTML parsing to finish. The `async` attribute
+does not work for inline classic scripts, but it does work for inline `<script type="module">`.
+
+### A note on file extensions
+
+You may have noticed we’re using the `.mjs` file extension for modules. On the Web, the file
+extension doesn’t really matter, as long as the file is served with [the JavaScript MIME type
+`text/javascript`](https://html.spec.whatwg.org/multipage/scripting.html#scriptingLanguages:javascript-mime-type).
+The browser knows it’s a module because of the `type` attribute on the script element.
+
+Still, we recommend using the `.mjs` extension for modules, for two reasons:
+
+1. During development, it makes it crystal clear that the file is a module as opposed to a
+    regular script. (It’s not always possible to tell just by looking at the code.) As mentioned
+    before, modules are treated differently than regular scripts, so the difference is hugely
+    important!
+2. It’s consistent with Node.js, where [the experimental modules implementation]
+    (https://nodejs.org/api/esm.html) only supports files with the `.mjs` extension.
```

Comment:
you can't have breaklines in links, the `)[` have to be on the same line.

Reply:
Good catch! Fixed.